operator/pkg/controllers/utils.go

```diff
@@ -471,3 +490,87 @@ func validateContainerResourceCustomizationValues(r customizev1beta1.ResourceReq
 	}
 	return nil
 }
+
+func ApplyContainerRateLimit(m *manifest.Objects, targetName string, containers []customizev1alpha1.ContainerReconcilerSpec) error {
+	if containers == nil || len(containers) == 0 {
+		return nil
+	}
+	// TODO: check for duplicated containers
+
+	// rateLimitMap is a map of container name to its corresponding rate limit customization.
+	rateLimitMap := make(map[string]*customizev1alpha1.RateLimit)
+	for _, c := range containers {
+		rateLimitMap[c.Name] = c.RateLimit
+	}
+	targetGVK := schema.GroupVersionKind{
+		Group:   appsv1.SchemeGroupVersion.Group,
+		Version: appsv1.SchemeGroupVersion.Version,
+		Kind:    "StatefulSet",
+	}
+	for _, item := range m.Items {
+		if item.GroupVersionKind() != targetGVK {
+			continue
+		}
+		if !strings.HasPrefix(item.GetName(), targetName) {
+			continue
+		}
+		if err := item.MutateContainers(customizeRateLimitFn(rateLimitMap)); err != nil {
+			return err
+		}
+		break // we already found the matching controller, no need to keep looking.
+	}
+
+	// check if all container resource customizations are applied
+	var notApplied []string
+	for c := range rateLimitMap {
+		notApplied = append(notApplied, c)
+	}
+	if len(notApplied) > 0 {
+		return fmt.Errorf("the following containers were not found in the manifest: %s", strings.Join(notApplied, ", "))
+	}
+	return nil
+}
+
+func customizeRateLimitFn(rateLimitMap map[string]*customizev1alpha1.RateLimit) func(container map[string]interface{}) error {
+	return func(container map[string]interface{}) error {
+		name, _, err := unstructured.NestedString(container, "name")
+		if err != nil {
+			return fmt.Errorf("error reading container name: %w", err)
+		}
+		rl, found := rateLimitMap[name]
+		if !found {
+			return nil
+		}
+		delete(rateLimitMap, name)
+		return applyRateLimitToContainerArg(container, rl)
+	}
+}
+
+func applyRateLimitToContainerArg(container map[string]interface{}, rateLimit *customizev1alpha1.RateLimit) error {
+	if rateLimit == nil {
+		return nil
+	}
+	origArgs, found, err := unstructured.NestedStringSlice(container, "args")
```

Comment:
Logic here is copied from #1988.

Reply:
Thank you! 
Have you checked if this works on a real CC manager container (I didn't have a e2e testing in real env)?

Reply:
I checked that the "--qps" and "--burst" are added to the real CC manager container args. See the PR description.